### ISSUE TITLE
Default field is not used when query is simple

### ISSUE THREAD

**Issue by AlxBystrov** [grindlemire/go-lucene#24]
Hi everybody!
Thanks for your parser, it's realy helps us to realize our goal - search log with lucene syntax with using clickhouse as backend.

In v0.0.16 relase you add ability to set up a defaultField option and it was great, but in my cases I found a little nuance: when query is simple, for example
example 1: "error"
example 2: "some string to search"

It's seems to be better to search this in defaultField

I placed in my forked repository an addition in a parse.go file with this addition (74 line)
```
			if final.Op == expr.Literal && p.defaultField != "" {
				final = expr.Expr(p.defaultField, expr.Equals, final.Left)
			}
```
This addition are closed my problem and I want to advise to use in your code.
Thanks again!

**Comment by grindlemire** [comment]
Thanks for this! This is indeed a bug in the code and your fix does fix it. I also added some tests to exercise this new behavior. Essentially there is an edge case I missed for single value literals not properly getting the default field. Your example 2 query of `some string to search` properly get's assigned the default field with no change but your example 1 query of `error` does not.

Let me think about where this edge case should be handled for the next couple days, it is possible you already have it in the proper position!

**Comment by grindlemire** [comment]
I merged a fix in with some tests and the fix should be in v0.20! Thanks for the report!
